chapters/packages.tex

```diff
@@ -326,3 +326,125 @@ \section{External resources}\label{external-resources}
+The files to support translation must be provided by the developer of the library.
+They must be stored in the resources directory \filename{modelica://$\mathit{LibraryName}$/Resources/Language/}.
+
+Two kind of files have to be provided:
+\begin{enumerate}
+\item Template file \filename{$\mathit{LibraryName}$.pot}, as the resource \filename{modelica://$\mathit{LibraryName}$/Resources/Language/$\mathit{LibraryName}$.pot}.
+It contains all necessary information to translate all descriptions, but no translations.
+The pattern \filename{$\mathit{LibraryName}$} denotes the toplevel class name of the library.
+\item One file for each supported language with the name \filename{$\mathit{LibraryName}$.$\mathit{language}$.po}, as the resource \filename{modelica://$\mathit{LibraryName}$/Resources/Language/$\mathit{LibraryName}$.$\mathit{language}$.po}.
+This file is a copy of the associated template file, but extended with the translations in the specified language.
+The pattern \filename{$\mathit{language}$} stands for the ISO 639-1 language code, e.g., \filename{de} or \filename{en}.
+\end{enumerate}
+
+The files consist of a header and a body.
+The detailed format of these files is described in \href{https://www.gnu.org/software/gettext/manual/gettext.pdf}{GNU gettext}.
+
+\begin{nonnormative}
+In the remainder of this section, several facts about the gettext specification are described non-normatively for easy access to some of the gettext basics.
+Always refer to the external gettext specification for full detail or in case of doubt.
+
+All text strings are in double quotes and encoded with UTF-8 characters.
+Comments start with an \lstinline!#! and are continued until the end of line.
+Spaces outside strings are ignored and used as separators.
+
+The header is marked with an empty \lstinline!msgid! entry and looks like this:
+\begin{lstlisting}
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: PACKAGE VERSION\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2019-03-15 10:52+0100\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"Language: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+\end{lstlisting}
+All general terms in the header should be replaced by specific information.
+
+In the body, directly following the header, for each descriptive text there is an entry of the form
+\begin{lstlisting}
+#: $\mathit{filename}$:$\mathit{lineNumber}$
+msgctxt $\mathit{messageContext}$
+msgid $\mathit{messageIdentifier}$
+msgstr $\mathit{messageTranslation}$
+\end{lstlisting}
+\end{nonnormative}
+
+Only the keywords \lstinline!msgctxt!, \lstinline!msgid! and \lstinline!msgstr! are used.
+
+\begin{nonnormative}
```

Comment:
To go with suggested change above:
```suggestion

```

Reply:
Now I see that there's is a connected problem here with references to the non-normatively introduced entry above.  Any chance we could resolve this more efficiently over zoom?